--- ../src-base/minecraft/net/minecraft/server/dedicated/DedicatedServer.java
+++ ../src-work/minecraft/net/minecraft/server/dedicated/DedicatedServer.java
@@ -71,7 +71,7 @@
 
     protected boolean func_71197_b() throws IOException {
         Thread thread = new Thread("Server console handler") {
-            private static final String __OBFID = "CL_00001786";
+        private static final String __OBFID = "CL_00001786";
             public void run() {
                 BufferedReader bufferedreader = new BufferedReader(new InputStreamReader(System.in));
                 String s4;
@@ -93,8 +93,10 @@
             field_155771_h.warn("To start the server with more ram, launch it as \"java -Xmx1024M -Xms1024M -jar minecraft_server.jar\"");
         }
 
+        jk_5.nailed.server.NailedEventFactory.fireServerStartBeforeConfig(this);
+
         field_155771_h.info("Loading properties");
-        this.field_71340_o = new PropertyManager(new File("server.properties"));
+        this.field_71340_o = new jk_5.nailed.server.config.MinecraftConfig();
         this.field_154332_n = new ServerEula(new File("eula.txt"));
 
         if (!this.field_154332_n.func_154346_a()) {
@@ -106,106 +108,47 @@
                 this.func_71189_e("127.0.0.1");
             } else {
                 this.func_71229_d(this.field_71340_o.func_73670_a("online-mode", true));
-                this.func_71189_e(this.field_71340_o.func_73671_a("server-ip", ""));
             }
 
-            this.func_71251_e(this.field_71340_o.func_73670_a("spawn-animals", true));
-            this.func_71257_f(this.field_71340_o.func_73670_a("spawn-npcs", true));
-            this.func_71188_g(this.field_71340_o.func_73670_a("pvp", true));
+            this.func_71251_e(true);
+            this.func_71257_f(true);
+            this.func_71188_g(true);
             this.func_71245_h(this.field_71340_o.func_73670_a("allow-flight", false));
-            this.func_180507_a_(this.field_71340_o.func_73671_a("resource-pack", ""), this.field_71340_o.func_73671_a("resource-pack-hash", ""));
             this.func_71205_p(this.field_71340_o.func_73671_a("motd", "A Minecraft Server"));
-            this.func_104055_i(this.field_71340_o.func_73670_a("force-gamemode", false));
+            this.func_104055_i(false);
             this.func_143006_e(this.field_71340_o.func_73669_a("player-idle-timeout", 0));
 
-            if (this.field_71340_o.func_73669_a("difficulty", 1) < 0) {
-                this.field_71340_o.func_73667_a("difficulty", Integer.valueOf(0));
-            } else if (this.field_71340_o.func_73669_a("difficulty", 1) > 3) {
-                this.field_71340_o.func_73667_a("difficulty", Integer.valueOf(3));
-            }
-
-            this.field_71338_p = this.field_71340_o.func_73670_a("generate-structures", true);
-            int i = this.field_71340_o.func_73669_a("gamemode", WorldSettings.GameType.SURVIVAL.func_77148_a());
-            this.field_71337_q = WorldSettings.func_77161_a(i);
+            this.field_71338_p = true;
+            this.field_71337_q = WorldSettings.GameType.ADVENTURE;
             field_155771_h.info("Default game type: " + this.field_71337_q);
-            InetAddress inetaddress = null;
 
-            if (this.func_71211_k().length() > 0) {
-                inetaddress = InetAddress.getByName(this.func_71211_k());
-            }
-
-            if (this.func_71215_F() < 0) {
-                this.func_71208_b(this.field_71340_o.func_73669_a("server-port", 25565));
-            }
-
             field_155771_h.info("Generating keypair");
             this.func_71253_a(CryptManager.func_75891_b());
             field_155771_h.info("Starting Minecraft server on " + (this.func_71211_k().length() == 0 ? "*" : this.func_71211_k()) + ":" + this.func_71215_F());
 
-            try {
-                this.func_147137_ag().func_151265_a(inetaddress, this.func_71215_F());
-            } catch (IOException ioexception) {
-                field_155771_h.warn("**** FAILED TO BIND TO PORT!");
-                field_155771_h.warn("The exception was: {}", new Object[] {ioexception.toString()});
-                field_155771_h.warn("Perhaps a server is already running on that port?");
-                return false;
-            }
+            jk_5.nailed.server.network.NailedNetworkManager.startEndpoints();
 
             if (!this.func_71266_T()) {
                 field_155771_h.warn("**** SERVER IS RUNNING IN OFFLINE/INSECURE MODE!");
                 field_155771_h.warn("The server will make no attempt to authenticate usernames. Beware.");
                 field_155771_h.warn("While this makes the game possible to play without internet access, it also opens up the ability for hackers to connect with any username they choose.");
-                field_155771_h.warn("To change this, set \"online-mode\" to \"true\" in the server.properties file.");
+                field_155771_h.warn("To change this, set \"online-mode\" to \"true\" in the settings.conf file.");
             }
 
-            if (this.func_152368_aE()) {
-                this.func_152358_ax().func_152658_c();
-            }
-
             if (!PreYggdrasilConverter.func_152714_a(this.field_71340_o)) {
                 return false;
             } else {
                 this.func_152361_a(new DedicatedPlayerList(this));
                 long j = System.nanoTime();
 
-                if (this.func_71270_I() == null) {
-                    this.func_71261_m(this.field_71340_o.func_73671_a("level-name", "world"));
-                }
-
-                String s = this.field_71340_o.func_73671_a("level-seed", "");
-                String s1 = this.field_71340_o.func_73671_a("level-type", "DEFAULT");
-                String s2 = this.field_71340_o.func_73671_a("generator-settings", "");
-                long k = (new Random()).nextLong();
-
-                if (s.length() > 0) {
-                    try {
-                        long l = Long.parseLong(s);
-
-                        if (l != 0L) {
-                            k = l;
-                        }
-                    } catch (NumberFormatException numberformatexception) {
-                        k = (long)s.hashCode();
-                    }
-                }
-
-                WorldType worldtype = WorldType.func_77130_a(s1);
-
-                if (worldtype == null) {
-                    worldtype = WorldType.field_77137_b;
-                }
-
                 this.func_147136_ar();
                 this.func_82356_Z();
                 this.func_110455_j();
                 this.func_70002_Q();
                 this.func_175577_aI();
-                this.func_71191_d(this.field_71340_o.func_73669_a("max-build-height", 256));
-                this.func_71191_d((this.func_71207_Z() + 8) / 16 * 16);
-                this.func_71191_d(MathHelper.func_76125_a(this.func_71207_Z(), 64, 256));
-                this.field_71340_o.func_73667_a("max-build-height", Integer.valueOf(this.func_71207_Z()));
-                field_155771_h.info("Preparing level \"" + this.func_71270_I() + "\"");
-                this.func_71247_a(this.func_71270_I(), this.func_71270_I(), k, worldtype, s2);
+                this.func_71191_d(256);
+                jk_5.nailed.server.NailedEventFactory.fireStartBeforeWorldLoad(this);
+                this.func_71247_a(this.func_71270_I(), this.func_71270_I(), 0, null, null);
                 long i1 = System.nanoTime() - j;
                 String s3 = String.format("%.3fs", new Object[] {Double.valueOf((double)i1 / 1.0E9D)});
                 field_155771_h.info("Done (" + s3 + ")! For help, type \"help\" or \"?\"");
@@ -229,6 +172,7 @@
                     thread1.start();
                 }
 
+                jk_5.nailed.server.NailedEventFactory.fireServerStarted(this);
                 return true;
             }
         }
@@ -248,11 +192,11 @@
     }
 
     public EnumDifficulty func_147135_j() {
-        return EnumDifficulty.func_151523_a(this.field_71340_o.func_73669_a("difficulty", 1));
+        return EnumDifficulty.NORMAL;
     }
 
     public boolean func_71199_h() {
-        return this.field_71340_o.func_73670_a("hardcore", false);
+        return false;
     }
 
     protected void func_71228_a(CrashReport p_71228_1_) {
@@ -295,11 +239,11 @@
     }
 
     public boolean func_71255_r() {
-        return this.field_71340_o.func_73670_a("allow-nether", true);
+        return true;
     }
 
     public boolean func_71193_K() {
-        return this.field_71340_o.func_73670_a("spawn-monsters", true);
+        return true;
     }
 
     public void func_70000_a(PlayerUsageSnooper p_70000_1_) {
@@ -370,11 +314,11 @@
     }
 
     public boolean func_82356_Z() {
-        return this.field_71340_o.func_73670_a("enable-command-block", false);
+        return true;
     }
 
     public int func_82357_ak() {
-        return this.field_71340_o.func_73669_a("spawn-protection", super.func_82357_ak());
+        return 0;
     }
 
     public boolean func_175579_a(World p_175579_1_, BlockPos p_175579_2_, EntityPlayer p_175579_3_) {
