--- ../src-base/minecraft/net/minecraft/server/dedicated/PropertyManager.java
+++ ../src-work/minecraft/net/minecraft/server/dedicated/PropertyManager.java
@@ -16,6 +16,7 @@
 
     public PropertyManager(File p_i45278_1_) {
         this.field_73673_c = p_i45278_1_;
+        if(p_i45278_1_ == null) return;
 
         if (p_i45278_1_.exists()) {
             FileInputStream fileinputstream = null;
