--- ../src-base/minecraft/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
+++ ../src-work/minecraft/net/minecraft/world/chunk/storage/AnvilChunkLoader.java
@@ -9,6 +9,10 @@
 import java.util.Iterator;
 import java.util.List;
 import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
 import net.minecraft.block.Block;
 import net.minecraft.entity.Entity;
 import net.minecraft.entity.EntityList;
@@ -24,8 +28,6 @@
 import net.minecraft.world.chunk.NibbleArray;
 import net.minecraft.world.storage.IThreadedFileIO;
 import net.minecraft.world.storage.ThreadedFileIOBase;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 
 public class AnvilChunkLoader implements IChunkLoader, IThreadedFileIO {
     private static final Logger field_151505_a = LogManager.getLogger();
@@ -39,16 +41,49 @@
         this.field_75825_d = p_i2003_1_;
     }
 
+    public boolean chunkExists(World world, int i, int j){
+        ChunkCoordIntPair chunkcoordintpair = new ChunkCoordIntPair(i, j);
+
+        synchronized(this.field_75827_c){
+            if(this.field_75826_b.contains(chunkcoordintpair)){
+                Iterator it = this.field_75828_a.iterator();
+                while(it.hasNext()){
+                    PendingChunk pendingChunk = (PendingChunk) it.next();
+                    if(pendingChunk.field_76548_a.equals(chunkcoordintpair)){
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return RegionFileCache.func_76550_a(this.field_75825_d, i, j).chunkExists(i & 31, j & 31);
+    }
+
     public Chunk func_75815_a(World p_75815_1_, int p_75815_2_, int p_75815_3_) throws IOException {
+        Object[] data = this.loadChunk__Async(p_75815_1_, p_75815_2_, p_75815_3_);
+
+        if(data != null){
+            Chunk chunk = (Chunk) data[0];
+            NBTTagCompound tag = (NBTTagCompound) data[1];
+            this.loadEntities(p_75815_1_, tag.func_74775_l("Level"), chunk);
+            return chunk;
+        }
+
+        return null;
+    }
+
+    public Object[] loadChunk__Async(World p_75815_1_, int p_75815_2_, int p_75815_3_) throws IOException {
         NBTTagCompound nbttagcompound = null;
         ChunkCoordIntPair chunkcoordintpair = new ChunkCoordIntPair(p_75815_2_, p_75815_3_);
         Object object = this.field_75827_c;
 
         synchronized (this.field_75827_c) {
             if (this.field_75826_b.contains(chunkcoordintpair)) {
-                for (int k = 0; k < this.field_75828_a.size(); ++k) {
-                    if (((AnvilChunkLoader.PendingChunk)this.field_75828_a.get(k)).field_76548_a.equals(chunkcoordintpair)) {
-                        nbttagcompound = ((AnvilChunkLoader.PendingChunk)this.field_75828_a.get(k)).field_76547_b;
+                Iterator it = this.field_75828_a.iterator();
+                while(it.hasNext()){
+                    PendingChunk pendingChunk = (PendingChunk) it.next();
+                    if(pendingChunk.field_76548_a.equals(chunkcoordintpair)){
+                        nbttagcompound = pendingChunk.field_76547_b;
                         break;
                     }
                 }
@@ -65,10 +100,21 @@
             nbttagcompound = CompressedStreamTools.func_74794_a(datainputstream);
         }
 
-        return this.func_75822_a(p_75815_1_, p_75815_2_, p_75815_3_, nbttagcompound);
+        return this.checkedReadChunkFromNBT__Async(p_75815_1_, p_75815_2_, p_75815_3_, nbttagcompound);
     }
 
     protected Chunk func_75822_a(World p_75822_1_, int p_75822_2_, int p_75822_3_, NBTTagCompound p_75822_4_) {
+        Object[] data = this.checkedReadChunkFromNBT__Async(p_75822_1_, p_75822_2_, p_75822_3_, p_75822_4_);
+
+        if(data != null){
+            Chunk chunk = (Chunk) data[0];
+            return chunk;
+        }
+
+        return null;
+    }
+
+    protected Object[] checkedReadChunkFromNBT__Async(World p_75822_1_, int p_75822_2_, int p_75822_3_, NBTTagCompound p_75822_4_) {
         if (!p_75822_4_.func_150297_b("Level", 10)) {
             field_151505_a.error("Chunk file at " + p_75822_2_ + "," + p_75822_3_ + " is missing level data, skipping");
             return null;
@@ -82,10 +128,26 @@
                 field_151505_a.error("Chunk file at " + p_75822_2_ + "," + p_75822_3_ + " is in the wrong location; relocating. (Expected " + p_75822_2_ + ", " + p_75822_3_ + ", got " + chunk.field_76635_g + ", " + chunk.field_76647_h + ")");
                 p_75822_4_.func_74768_a("xPos", p_75822_2_);
                 p_75822_4_.func_74768_a("zPos", p_75822_3_);
+                // Have to move tile entities since we don't load them at this stage
+                NBTTagList tileEntities = p_75822_4_.func_74775_l("Level").func_150295_c("TileEntities", 10);
+
+                if(tileEntities != null){
+                    for(int te = 0; te < tileEntities.func_74745_c(); te++){
+                        NBTTagCompound tileEntity = (NBTTagCompound) tileEntities.func_150305_b(te);
+                        int x = tileEntity.func_74762_e("x") - chunk.field_76635_g * 16;
+                        int z = tileEntity.func_74762_e("z") - chunk.field_76647_h * 16;
+                        tileEntity.func_74768_a("x", p_75822_2_ * 16 + x);
+                        tileEntity.func_74768_a("z", p_75822_3_ * 16 + z);
+                    }
+                }
+
                 chunk = this.func_75823_a(p_75822_1_, p_75822_4_.func_74775_l("Level"));
             }
 
-            return chunk;
+            Object[] data = new Object[2];
+            data[0] = chunk;
+            data[1] = p_75822_4_;
+            return data;
         }
     }
 
@@ -271,20 +333,20 @@
         ExtendedBlockStorage[] aextendedblockstorage = new ExtendedBlockStorage[b0];
         boolean flag = !p_75823_1_.field_73011_w.field_76576_e;
 
-        for (int k = 0; k < nbttaglist.func_74745_c(); ++k) {
+        for(int k = 0; k < nbttaglist.func_74745_c(); ++k){
             NBTTagCompound nbttagcompound1 = nbttaglist.func_150305_b(k);
             byte b1 = nbttagcompound1.func_74771_c("Y");
             ExtendedBlockStorage extendedblockstorage = new ExtendedBlockStorage(b1 << 4, flag);
             extendedblockstorage.func_76664_a(nbttagcompound1.func_74770_j("Blocks"));
 
-            if (nbttagcompound1.func_150297_b("Add", 7)) {
+            if(nbttagcompound1.func_150297_b("Add", 7)){
                 extendedblockstorage.func_76673_a(new NibbleArray(nbttagcompound1.func_74770_j("Add"), 4));
             }
 
             extendedblockstorage.func_76668_b(new NibbleArray(nbttagcompound1.func_74770_j("Data"), 4));
             extendedblockstorage.func_76659_c(new NibbleArray(nbttagcompound1.func_74770_j("BlockLight"), 4));
 
-            if (flag) {
+            if(flag){
                 extendedblockstorage.func_76666_d(new NibbleArray(nbttagcompound1.func_74770_j("SkyLight"), 4));
             }
 
@@ -294,10 +356,15 @@
 
         chunk.func_76602_a(aextendedblockstorage);
 
-        if (p_75823_2_.func_150297_b("Biomes", 7)) {
+        if(p_75823_2_.func_150297_b("Biomes", 7)){
             chunk.func_76616_a(p_75823_2_.func_74770_j("Biomes"));
         }
 
+        // End this method here and split off entity loading to another method
+        return chunk;
+    }
+
+    public void loadEntities(World p_75823_1_, NBTTagCompound p_75823_2_, Chunk chunk){
         NBTTagList nbttaglist1 = p_75823_2_.func_150295_c("Entities", 10);
 
         if (nbttaglist1 != null) {
@@ -348,7 +415,7 @@
             }
         }
 
-        return chunk;
+        //return chunk;
     }
 
     static class PendingChunk {
