--- ../src-base/minecraft/net/minecraft/server/management/ItemInWorldManager.java
+++ ../src-work/minecraft/net/minecraft/server/management/ItemInWorldManager.java
@@ -5,12 +5,13 @@
 import net.minecraft.entity.player.EntityPlayer;
 import net.minecraft.entity.player.EntityPlayerMP;
 import net.minecraft.item.ItemStack;
-import net.minecraft.item.ItemSword;
 import net.minecraft.network.play.server.S23PacketBlockChange;
 import net.minecraft.world.World;
 import net.minecraft.world.WorldServer;
 import net.minecraft.world.WorldSettings;
 
+import jk_5.nailed.server.NailedEventFactory;
+
 public class ItemInWorldManager {
     public World field_73092_a;
     public EntityPlayerMP field_73090_b;
@@ -177,10 +178,8 @@
     }
 
     public boolean func_73084_b(int p_73084_1_, int p_73084_2_, int p_73084_3_) {
-        if (this.field_73091_c.func_82752_c() && !this.field_73090_b.func_82246_f(p_73084_1_, p_73084_2_, p_73084_3_)) {
+        if(NailedEventFactory.fireOnBlockBroken(this.field_73092_a, this.field_73091_c, this.field_73090_b, p_73084_1_, p_73084_2_, p_73084_3_)){
             return false;
-        } else if (this.field_73091_c.func_77145_d() && this.field_73090_b.func_70694_bm() != null && this.field_73090_b.func_70694_bm().func_77973_b() instanceof ItemSword) {
-            return false;
         } else {
             Block block = this.field_73092_a.func_147439_a(p_73084_1_, p_73084_2_, p_73084_3_);
             int l = this.field_73092_a.func_72805_g(p_73084_1_, p_73084_2_, p_73084_3_);
@@ -241,6 +240,9 @@
     }
 
     public boolean func_73078_a(EntityPlayer p_73078_1_, World p_73078_2_, ItemStack p_73078_3_, int p_73078_4_, int p_73078_5_, int p_73078_6_, int p_73078_7_, float p_73078_8_, float p_73078_9_, float p_73078_10_) {
+        if(NailedEventFactory.fireOnRightClick(p_73078_1_, p_73078_2_, p_73078_3_, p_73078_4_, p_73078_5_, p_73078_6_, p_73078_7_, p_73078_8_, p_73078_9_, p_73078_10_)){
+            return false;
+        }
         if ((!p_73078_1_.func_70093_af() || p_73078_1_.func_70694_bm() == null) && p_73078_2_.func_147439_a(p_73078_4_, p_73078_5_, p_73078_6_).func_149727_a(p_73078_2_, p_73078_4_, p_73078_5_, p_73078_6_, p_73078_1_, p_73078_7_, p_73078_8_, p_73078_9_, p_73078_10_)) {
             return true;
         } else if (p_73078_3_ == null) {
