--- ../src-base/minecraft/net/minecraft/server/management/ItemInWorldManager.java
+++ ../src-work/minecraft/net/minecraft/server/management/ItemInWorldManager.java
@@ -177,10 +177,8 @@
     }
 
     public boolean func_73084_b(int p_73084_1_, int p_73084_2_, int p_73084_3_) {
-        if (this.field_73091_c.func_82752_c() && !this.field_73090_b.func_82246_f(p_73084_1_, p_73084_2_, p_73084_3_)) {
+        if(jk_5.nailed.server.NailedEventFactory.fireOnBlockBroken(this.field_73092_a, this.field_73091_c, this.field_73090_b, p_73084_1_, p_73084_2_, p_73084_3_)){
             return false;
-        } else if (this.field_73091_c.func_77145_d() && this.field_73090_b.func_70694_bm() != null && this.field_73090_b.func_70694_bm().func_77973_b() instanceof ItemSword) {
-            return false;
         } else {
             Block block = this.field_73092_a.func_147439_a(p_73084_1_, p_73084_2_, p_73084_3_);
             int l = this.field_73092_a.func_72805_g(p_73084_1_, p_73084_2_, p_73084_3_);
@@ -211,6 +209,7 @@
     }
 
     public boolean func_73085_a(EntityPlayer p_73085_1_, World p_73085_2_, ItemStack p_73085_3_) {
+        if(jk_5.nailed.server.NailedEventFactory.fireOnItemRightClick(p_73085_1_, p_73085_2_, p_73085_3_)) return false;
         int i = p_73085_3_.field_77994_a;
         int j = p_73085_3_.func_77960_j();
         ItemStack itemstack1 = p_73085_3_.func_77957_a(p_73085_2_, p_73085_1_);
