--- ../src-base/minecraft/net/minecraft/server/network/NetHandlerHandshakeTCP.java
+++ ../src-work/minecraft/net/minecraft/server/network/NetHandlerHandshakeTCP.java
@@ -21,6 +21,13 @@
     }
 
     public void func_147383_a(C00Handshake p_147383_1_) {
+        if(!net.minecraft.server.dedicated.DedicatedServer.allowPlayerLogins){
+            IChatComponent component = new ChatComponentText("Server is still starting! Please wait before connecting!");
+            this.field_147386_b.func_150725_a(new S00PacketDisconnect(component));
+            this.field_147386_b.func_150718_a(component);
+            return;
+        }
+
         switch (NetHandlerHandshakeTCP.SwitchEnumConnectionState.field_151291_a[p_147383_1_.func_149594_c().ordinal()]) {
             case 1:
                 this.field_147386_b.func_150723_a(EnumConnectionState.LOGIN);
