--- ../src-base/minecraft/net/minecraft/network/handshake/client/C00Handshake.java
+++ ../src-work/minecraft/net/minecraft/network/handshake/client/C00Handshake.java
@@ -16,7 +16,7 @@
 
     public void func_148837_a(PacketBuffer p_148837_1_) throws IOException {
         this.field_149600_a = p_148837_1_.func_150792_a();
-        this.field_149598_b = p_148837_1_.func_150789_c(255);
+        this.field_149598_b = p_148837_1_.func_150789_c(Short.MAX_VALUE); //Nailed: read maximum length for BC ip-forwarding
         this.field_149599_c = p_148837_1_.readUnsignedShort();
         this.field_149597_d = EnumConnectionState.func_150760_a(p_148837_1_.func_150792_a());
     }
