--- ../src-base/minecraft/net/minecraft/world/storage/SaveHandler.java
+++ ../src-work/minecraft/net/minecraft/world/storage/SaveHandler.java
@@ -28,7 +28,7 @@
     public SaveHandler(File p_i2146_1_, String p_i2146_2_, boolean p_i2146_3_) {
         this.field_75770_b = new File(p_i2146_1_, p_i2146_2_);
         this.field_75770_b.mkdirs();
-        this.field_75771_c = new File(this.field_75770_b, "playerdata");
+        this.field_75771_c = new File("playerdata");
         this.field_75768_d = new File(this.field_75770_b, "data");
         this.field_75768_d.mkdirs();
         this.field_75767_f = p_i2146_2_;
