--- ../src-base/minecraft/net/minecraft/world/GameRules.java
+++ ../src-work/minecraft/net/minecraft/world/GameRules.java
@@ -10,6 +10,7 @@
     private static final String __OBFID = "CL_00000136";
 
     public GameRules() {
+        if(true) return; //Nailed: don't initialize them here. See DefaultGameRules
         this.func_82769_a("doFireTick", "true");
         this.func_82769_a("mobGriefing", "true");
         this.func_82769_a("keepInventory", "false");
