--- ../src-base/minecraft/net/minecraft/world/GameRules.java
+++ ../src-work/minecraft/net/minecraft/world/GameRules.java
@@ -10,6 +10,7 @@
     private static final String __OBFID = "CL_00000136";
 
     public GameRules() {
+        if(true) return; //Nailed: don't initialize them here. See DefaultGameRules
         this.func_180262_a("doFireTick", "true", GameRules.ValueType.BOOLEAN_VALUE);
         this.func_180262_a("mobGriefing", "true", GameRules.ValueType.BOOLEAN_VALUE);
         this.func_180262_a("keepInventory", "false", GameRules.ValueType.BOOLEAN_VALUE);
