--- ../src-base/minecraft/net/minecraft/world/WorldProvider.java
+++ ../src-work/minecraft/net/minecraft/world/WorldProvider.java
@@ -11,6 +11,7 @@
 import net.minecraft.world.gen.ChunkProviderFlat;
 import net.minecraft.world.gen.ChunkProviderGenerate;
 import net.minecraft.world.gen.FlatGeneratorInfo;
+import net.minecraft.world.storage.WorldInfo;
 
 public abstract class WorldProvider {
     public static final float[] field_111203_a = new float[] {1.0F, 0.75F, 0.5F, 0.25F, 0.0F, 0.25F, 0.5F, 0.75F};
@@ -94,7 +95,7 @@
     }
 
     public static WorldProvider func_76570_a(int p_76570_0_) {
-        return (WorldProvider)(p_76570_0_ == -1 ? new WorldProviderHell() : (p_76570_0_ == 0 ? new WorldProviderSurface() : (p_76570_0_ == 1 ? new WorldProviderEnd() : null)));
+        return jk_5.nailed.server.world.NailedDimensionManager.instance().createProviderFor(p_76570_0_);
     }
 
     public BlockPos func_177496_h() {
@@ -107,6 +108,19 @@
 
     public abstract String func_80007_l();
 
+    /**
+     * ****** Nailed Start ********
+     */
+
+    public void setDimension(int dimensionId){
+        this.field_76574_g = dimensionId;
+    }
+
+    public jk_5.nailed.api.util.Location getSpawnPoint(){
+        WorldInfo info = this.field_76579_a.field_72986_A;
+        return new jk_5.nailed.api.util.Location(null, info.func_76079_c(), info.func_76075_d(), info.func_76074_e(), 0, 0);
+    }
+
     public abstract String func_177498_l();
 
     public WorldChunkManager func_177499_m() {
