--- ../src-base/minecraft/net/minecraft/world/WorldServer.java
+++ ../src-work/minecraft/net/minecraft/world/WorldServer.java
@@ -105,15 +105,12 @@
 
         scoreboardsavedata.func_96499_a(this.field_96442_D);
         ((ServerScoreboard)this.field_96442_D).func_96547_a(scoreboardsavedata);
+        jk_5.nailed.server.world.NailedDimensionManager.setWorld(p_i45284_4_, this);
     }
 
     public void func_72835_b() {
         super.func_72835_b();
 
-        if (this.func_72912_H().func_76093_s() && this.field_73013_u != EnumDifficulty.HARD) {
-            this.field_73013_u = EnumDifficulty.HARD;
-        }
-
         this.field_73011_w.field_76578_c.func_76938_b();
 
         if (this.func_73056_e()) {
