--- ../src-base/minecraft/net/minecraft/world/World.java
+++ ../src-work/minecraft/net/minecraft/world/World.java
@@ -2333,13 +2333,8 @@
     }
 
     public BlockPos func_175694_M() {
-        BlockPos blockpos = new BlockPos(this.field_72986_A.func_76079_c(), this.field_72986_A.func_76075_d(), this.field_72986_A.func_76074_e());
-
-        if (!this.func_175723_af().func_177746_a(blockpos)) {
-            blockpos = this.func_175645_m(new BlockPos(this.func_175723_af().func_177731_f(), 0.0D, this.func_175723_af().func_177721_g()));
-        }
-
-        return blockpos;
+        jk_5.nailed.api.util.Location spawn = this.field_73011_w.getSpawnPoint();
+        return new BlockPos(spawn.getFloorX(), spawn.getFloorY(), spawn.getFloorZ());
     }
 
     public void func_175652_B(BlockPos p_175652_1_) {
@@ -2369,7 +2364,7 @@
     }
 
     public GameRules func_82736_K() {
-        return this.field_72986_A.func_82574_x();
+        return jk_5.nailed.server.map.gamerule.DelegatingEditableGameRules.get(((jk_5.nailed.api.world.World) this).getGamerules());
     }
 
     public void func_72854_c() {}
