--- ../src-base/minecraft/net/minecraft/block/BlockCommandBlock.java
+++ ../src-work/minecraft/net/minecraft/block/BlockCommandBlock.java
@@ -10,6 +10,8 @@
 import net.minecraft.tileentity.TileEntityCommandBlock;
 import net.minecraft.world.World;
 
+import jk_5.nailed.server.tileentity.TileEntityStatEmitter;
+
 public class BlockCommandBlock extends BlockContainer {
     private static final String __OBFID = "CL_00000219";
 
@@ -18,6 +20,9 @@
     }
 
     public TileEntity func_149915_a(World p_149915_1_, int p_149915_2_) {
+        if(p_149915_2_ == 8){
+            return new TileEntityStatEmitter();
+        }
         return new TileEntityCommandBlock();
     }
 
