--- ../src-base/minecraft/net/minecraft/network/NetHandlerPlayServer.java
+++ ../src-work/minecraft/net/minecraft/network/NetHandlerPlayServer.java
@@ -319,7 +319,7 @@
 
                 AxisAlignedBB axisalignedbb = this.field_147369_b.field_70121_D.func_72329_c().func_72314_b((double)f3, (double)f3, (double)f3).func_72321_a(0.0D, -0.55D, 0.0D);
 
-                if (!this.field_147367_d.func_71231_X() && !this.field_147369_b.field_71134_c.func_73083_d() && !worldserver.func_72829_c(axisalignedbb)) {
+                if (!this.field_147367_d.func_71231_X() && !this.field_147369_b.field_71075_bZ.field_75101_c && !worldserver.func_72829_c(axisalignedbb)) {
                     if (d11 >= -0.03125D) {
                         ++this.field_147365_f;
 
@@ -356,9 +356,9 @@
         this.field_147369_b.func_143004_u();
 
         if (p_147345_1_.func_149506_g() == 4) {
-            this.field_147369_b.func_71040_bB(false);
+            if(jk_5.nailed.server.NailedEventFactory.firePlayerDropStack(this.field_147369_b, false)) this.field_147369_b.func_71040_bB(false);
         } else if (p_147345_1_.func_149506_g() == 3) {
-            this.field_147369_b.func_71040_bB(true);
+            if(jk_5.nailed.server.NailedEventFactory.firePlayerDropStack(this.field_147369_b, true)) this.field_147369_b.func_71040_bB(true);
         } else if (p_147345_1_.func_149506_g() == 5) {
             this.field_147369_b.func_71034_by();
         } else {
@@ -499,9 +499,7 @@
     public void func_147231_a(IChatComponent p_147231_1_) {
         field_147370_c.info(this.field_147369_b.func_70005_c_() + " lost connection: " + p_147231_1_);
         this.field_147367_d.func_147132_au();
-        ChatComponentTranslation chatcomponenttranslation = new ChatComponentTranslation("multiplayer.player.left", new Object[] {this.field_147369_b.func_145748_c_()});
-        chatcomponenttranslation.func_150256_b().func_150238_a(EnumChatFormatting.YELLOW);
-        this.field_147367_d.func_71203_ab().func_148539_a(chatcomponenttranslation);
+        jk_5.nailed.server.NailedEventFactory.firePlayerLeft(this.field_147369_b);
         this.field_147369_b.func_71123_m();
         this.field_147367_d.func_71203_ab().func_72367_e(this.field_147369_b);
 
@@ -569,8 +567,13 @@
             if (s.startsWith("/")) {
                 this.func_147361_d(s);
             } else {
-                ChatComponentTranslation chatcomponenttranslation1 = new ChatComponentTranslation("chat.type.text", new Object[] {this.field_147369_b.func_145748_c_(), s});
-                this.field_147367_d.func_71203_ab().func_148544_a(chatcomponenttranslation1, false);
+                //Nailed start
+                String ret = jk_5.nailed.server.NailedEventFactory.firePlayerChat(this.field_147369_b, p_147354_1_.func_149439_c());
+                if(ret != null){
+                    ChatComponentTranslation chatcomponenttranslation1 = new ChatComponentTranslation("chat.type.text", this.field_147369_b.func_145748_c_(), ret);
+                    this.field_147367_d.func_71203_ab().func_148544_a(chatcomponenttranslation1, false);
+                }
+                //Nailed end
             }
 
             this.field_147374_l += 20;
@@ -651,6 +654,8 @@
 
         switch (NetHandlerPlayServer.SwitchEnumState.field_151290_a[enumstate.ordinal()]) {
             case 1:
+                jk_5.nailed.server.NailedEventFactory.onPlayerRespawn(this.field_147369_b);
+                if(true) break;
                 if (this.field_147369_b.field_71136_j) {
                     this.field_147369_b = this.field_147367_d.func_71203_ab().func_72368_a(this.field_147369_b, 0, true);
                 } else if (this.field_147369_b.func_71121_q().func_72912_H().func_76093_s()) {
