--- ../src-base/minecraft/net/minecraft/network/NetworkManager.java
+++ ../src-work/minecraft/net/minecraft/network/NetworkManager.java
@@ -61,6 +61,8 @@
     private boolean field_152463_r;
     private boolean field_179297_n;
     private static final String __OBFID = "CL_00001240";
+    public java.util.UUID spoofedUUID;
+    public com.mojang.authlib.properties.Property[] spoofedProfile;
 
     public NetworkManager(EnumPacketDirection p_i46004_1_) {
         this.field_179294_g = p_i46004_1_;
