--- ../src-base/minecraft/net/minecraft/item/ItemStack.java
+++ ../src-work/minecraft/net/minecraft/item/ItemStack.java
@@ -103,6 +103,7 @@
     }
 
     public boolean func_179546_a(EntityPlayer p_179546_1_, World p_179546_2_, BlockPos p_179546_3_, EnumFacing p_179546_4_, float p_179546_5_, float p_179546_6_, float p_179546_7_) {
+        if(jk_5.nailed.server.NailedEventFactory.fireOnRightClick(p_179546_1_, p_179546_2_, this, p_179546_3_, p_179546_4_, p_179546_5_, p_179546_6_, p_179546_7_)) return false;
         boolean flag = this.func_77973_b().func_180614_a(this, p_179546_1_, p_179546_2_, p_179546_3_, p_179546_4_, p_179546_5_, p_179546_6_, p_179546_7_);
 
         if (flag) {
