--- ../src-base/minecraft/net/minecraft/command/CommandHandler.java
+++ ../src-work/minecraft/net/minecraft/command/CommandHandler.java
@@ -25,6 +25,11 @@
     public int func_71556_a(ICommandSender p_71556_1_, String p_71556_2_) {
         p_71556_2_ = p_71556_2_.trim();
 
+        //Nailed start: Attempt to handle the command with our system first
+        int result = jk_5.nailed.server.NailedEventFactory.fireCommand(p_71556_1_, p_71556_2_);
+        if(result != -1) return result;
+        //Nailed end
+
         if (p_71556_2_.startsWith("/")) {
             p_71556_2_ = p_71556_2_.substring(1);
         }
@@ -126,6 +131,11 @@
         String[] astring = p_180524_2_.split(" ", -1);
         String s1 = astring[0];
 
+        //Nailed start
+        List<String> ret = jk_5.nailed.server.NailedEventFactory.fireTabCompletion(p_71558_1_, p_71558_2_);
+        if(ret != null) return ret;
+        //Nailed end
+
         if (astring.length == 1) {
             ArrayList arraylist = Lists.newArrayList();
             Iterator iterator = this.field_71562_a.entrySet().iterator();
