--- ../src-base/minecraft/net/minecraft/network/play/server/S02PacketChat.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/S02PacketChat.java
@@ -9,6 +9,7 @@
 
 public class S02PacketChat extends Packet {
     private IChatComponent field_148919_a;
+    private jk_5.nailed.api.chat.BaseComponent[] comps;
     private boolean field_148918_b;
     private static final String __OBFID = "CL_00001289";
 
@@ -26,12 +27,25 @@
         this.field_148918_b = p_i45180_2_;
     }
 
+    public S02PacketChat(jk_5.nailed.api.chat.BaseComponent... comps){
+        this(true, comps);
+    }
+
+    public S02PacketChat(boolean isChat, jk_5.nailed.api.chat.BaseComponent... comps){
+        this.comps = comps;
+        this.field_148918_b = isChat;
+    }
+
     public void func_148837_a(PacketBuffer p_148837_1_) throws IOException {
         this.field_148919_a = IChatComponent.Serializer.func_150699_a(p_148837_1_.func_150789_c(32767));
     }
 
     public void func_148840_b(PacketBuffer p_148840_1_) throws IOException {
-        p_148840_1_.func_150785_a(IChatComponent.Serializer.func_150696_a(this.field_148919_a));
+        if(comps == null){
+            p_148840_1_.func_150785_a(IChatComponent.Serializer.func_150696_a(this.field_148919_a));
+        }else{
+            p_148840_1_.func_150785_a(jk_5.nailed.api.chat.serialization.ComponentSerializer.toString(this.comps));
+        }
     }
 
     public void func_148833_a(INetHandlerPlayClient p_148833_1_) {
