--- ../src-base/minecraft/net/minecraft/network/play/server/S02PacketChat.java
+++ ../src-work/minecraft/net/minecraft/network/play/server/S02PacketChat.java
@@ -9,6 +9,7 @@
 
 public class S02PacketChat implements Packet {
     private IChatComponent field_148919_a;
+    private jk_5.nailed.api.chat.BaseComponent[] comps;
     private byte field_179842_b;
     private static final String __OBFID = "CL_00001289";
 
@@ -18,18 +19,31 @@
         this(p_i45179_1_, (byte)1);
     }
 
+    public S02PacketChat(jk_5.nailed.api.chat.BaseComponent... comps){
+        this((byte)1, comps);
+    }
+
     public S02PacketChat(IChatComponent p_i45986_1_, byte p_i45986_2_) {
         this.field_148919_a = p_i45986_1_;
         this.field_179842_b = p_i45986_2_;
     }
 
+    public S02PacketChat(byte type, jk_5.nailed.api.chat.BaseComponent... comps){
+        this.field_179842_b = type;
+        this.comps = comps;
+    }
+
     public void func_148837_a(PacketBuffer p_148837_1_) throws IOException {
         this.field_148919_a = p_148837_1_.func_179258_d();
         this.field_179842_b = p_148837_1_.readByte();
     }
 
     public void func_148840_b(PacketBuffer p_148840_1_) throws IOException {
-        p_148840_1_.func_179256_a(this.field_148919_a);
+        if(comps == null){
+            p_148840_1_.func_180714_a(IChatComponent.Serializer.func_150696_a(this.field_148919_a));
+        }else{
+            p_148840_1_.func_180714_a(jk_5.nailed.api.chat.serialization.ComponentSerializer.toString(this.comps));
+        }
         p_148840_1_.writeByte(this.field_179842_b);
     }
 
